src/interpreter/token.rs

```diff
@@ -0,0 +1,8 @@
+// split to iterator.
+use std::str;
+
+type MyIter<'a> = str::SplitWhitespace<'a>; //type aliasing
```

Comment:
MyIter -> TokenIterator
comments should also have whitespace. e.g. `// type aliasing` vs. `//type aliasing`, and be complete sentences used for clarification.